### ISSUE TITLE
the random seed for parameter selection and step data

### ISSUE THREAD

**Issue by stylebased** [ojwalch/sleep_classifiers#23]
Hi,
I am really impressed by your paper. When I tried to reproduce your results with your codes and dataset, the results looks different from your results in paper eg. Figure4. Can I assume this difference is caused by randomness in hyperparameter-selection? If so, what is the random seed you used that I can exactly reproduce your results?
Best

**Comment by ojwalch** [comment]
Hi! Tragically, I didn't keep track of the random seed when I was doing this work—my bad! How different are the plots looking? Feel free to reach out at ojwalch @ umich.edu if you'd like to follow-up there. 

Best,
Olivia

**Comment by stylebased** [comment]
Thank you for your quick reply.This is the figure 3 in your paper<img width="351" alt="image" src="https://user-images.githubusercontent.com/46028440/177872058-79410510-021a-48a5-8cfc-c4fc4ba4686f.png">. This is the plot I got with same title 
<img width="364" alt="image" src="https://user-images.githubusercontent.com/46028440/177872267-8f4e3e26-fc85-4ce6-b5a3-72a967536a3c.png">
.
Does this difference make sense? Did you use Monte Carlo hyper-parameter searching method?

**Comment by ojwalch** [comment]
Yep, that difference makes sense: For those figures, I'm randomly choosing the training and testing set, evaluating performance, and then repeating the process many times. The random choices of training/testing set will cause variation in this figure, as will the number of times you repeat the simulation (I believe I did 200 times for the paper). Hope this helps!

**Comment by stylebased** [comment]
Your response is really helpful. I notice there are two hyper-parameter selection method in your codes. One is Leave-One-Out and another is Monte Carlo. Which one did you use in your paper figures such as figure 2, figure 3. Thanks

**Comment by ojwalch** [comment]
Both Figures 2 and 3 are Monte Carlo cross validation 👍

**Comment by stylebased** [comment]
thanks

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Olivia Walch ***@***.***>
发送时间: Thursday, July 7, 2022 9:13:54 PM
收件人: ojwalch/sleep_classifiers ***@***.***>
抄送: Hu, Xiangyu ***@***.***>; Author ***@***.***>
主题: Re: [ojwalch/sleep_classifiers] the random seed for parameter selection and step data (Issue #23)


Both Figures 2 and 3 are Monte Carlo cross validation 👍

**Comment by stylebased** [comment]
I have another question. In your 'clock proxy' and time-based feature part, you used the steps to estimate the light. Can you provide me more detail things about how did you use steps data to predict the circadian clock? For example, what is the threshold value you set to say there is light? In your paper, you said 'if steps were above a threshold, the light was assumed to be one of three levels'
